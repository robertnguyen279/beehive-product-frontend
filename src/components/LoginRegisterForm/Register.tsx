import React from 'react';
import { Input } from 'components/Forms';
import UserIcon from 'assets/icons/user.svg';
import PasswordIcon from 'assets/icons/key.svg';
import Button from 'components/Button';
import { useTranslation } from 'react-i18next';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import { LoginRegisterForm } from 'types/main';

const Register = ({ onChange }: LoginRegisterForm): React.ReactElement => {
  const { t } = useTranslation(['Common']);
  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
      confirmPassword: '',
    },
    validationSchema: Yup.object({
      email: Yup.string().email(t('Common:form-email-error')).required(t('Common:form-required')),
      password: Yup.string().required(t('Common:form-required')),
      confirmPassword: Yup.string()
        .required(t('Common:form-required'))
        .oneOf([Yup.ref('password'), null], t('Common:form-password-confirm-error')),
    }),
    onSubmit: (value) => {
      console.log(value);
    },
  });
  return (
    <form
      className="w-full flex flex-col items-center justify-center animate__animated animate__fadeInRight"
      method="POST"
      onSubmit={formik.handleSubmit}
    >
      <Input
        value={formik.values.email}
        onChange={formik.handleChange}
        type="text"
        placeholder={t('Common:form-email-placeholder')}
        name="email"
        className="mb-3 w-4/6"
        prefixIcon={UserIcon}
        error={formik.errors.email && formik.touched.email ? formik.errors.email : false}
      />
      <Input
        value={formik.values.password}
        onChange={formik.handleChange}
        type="password"
        placeholder={t('Common:form-password-placeholder')}
        name="password"
        className="mb-3 w-4/6"
        prefixIcon={PasswordIcon}
        error={formik.errors.password && formik.touched.password ? formik.errors.password : false}
      />
      <Input
        value={formik.values.confirmPassword}
        onChange={formik.handleChange}
        type="password"
        placeholder={t('Common:form-password-confirm-placeholder')}
        name="confirmPassword"
        className="mb-3 w-4/6"
        prefixIcon={PasswordIcon}
        error={formik.errors.confirmPassword && formik.touched.confirmPassword ? formik.errors.confirmPassword : false}
      />
      <Button className="w-4/6 mb-2" type="submit">
        {t('Common:button-signup')}
      </Button>
      <div className="text-sm text-primary cursor-pointer" onClick={() => onChange()}>
        {t('Common:form-or-login')}
      </div>
      ;
    </form>
  );
};

export default Register;
