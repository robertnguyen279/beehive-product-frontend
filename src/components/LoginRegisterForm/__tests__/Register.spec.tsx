import React from 'react';
import { shallow } from 'enzyme';

import Register from '../Register';
// const clickFn = jest.fn();

describe('Register', () => {
  const component = shallow(
    <Register
      onChange={() => {
        console.log('change');
      }}
    />,
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
