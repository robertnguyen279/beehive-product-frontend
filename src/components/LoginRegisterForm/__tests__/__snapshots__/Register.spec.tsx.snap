// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
<form
  className="w-full flex flex-col items-center justify-center animate__animated animate__fadeInRight"
  method="POST"
  onSubmit={[Function]}
>
  <Input
    className="mb-3 w-4/6"
    error={false}
    name="email"
    onChange={[Function]}
    placeholder="Common:form-email-placeholder"
    prefixIcon="test-file-stub"
    type="text"
    value=""
  />
  <Input
    className="mb-3 w-4/6"
    error={false}
    name="password"
    onChange={[Function]}
    placeholder="Common:form-password-placeholder"
    prefixIcon="test-file-stub"
    type="password"
    value=""
  />
  <Input
    className="mb-3 w-4/6"
    error={false}
    name="confirmPassword"
    onChange={[Function]}
    placeholder="Common:form-password-confirm-placeholder"
    prefixIcon="test-file-stub"
    type="password"
    value=""
  />
  <Button
    className="w-4/6 mb-2"
    type="submit"
  >
    Common:button-signup
  </Button>
  <div
    className="text-sm text-primary cursor-pointer"
    onClick={[Function]}
  >
    Common:form-or-login
  </div>
</form>
`;
